src/controllers/csi/gc/binaries.go

```diff
@@ -27,7 +27,7 @@ func (gc *CSIGarbageCollector) runBinaryGarbageCollection(tenantUUID string, lat
 
 	for _, version := range storedVersions {
 		shouldDelete := isNotLatestVersion(version, latestVersion) &&
-			shouldDeleteVersion(version, usedVersions)
+			shouldDeleteVersion(version, usedVersions) && !pinnedVersions[version]
```

Comment:
nitpick: since `pinnedVersionSet` is already a type, it could be extended with a method such as `isNotPinned(version string)` which could be used in this condition to enhance readability.

```suggestion
			shouldDeleteVersion(version, usedVersions) && pinnedVersions.isNotPinned(version)
```

Reply:
done